deltachat-ios/AppDelegate.swift

```diff
@@ -435,19 +428,23 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
     // we have 30 seconds time for our job, things are quite similar as in `didReceiveRemoteNotification`
     func application(_ application: UIApplication, performFetchWithCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
         logger.info("➡️ Notifications: performFetchWithCompletionHandler")
-        increaseDebugCounter("notify-local-wakeup")
-        pushToDebugArray("🏠")
+        UserDefaults.pushToDebugArray("🏠")
         performFetch(completionHandler: completionHandler)
     }
 
     private func performFetch(completionHandler: ((UIBackgroundFetchResult) -> Void)? = nil) {
-        NotificationManager.removeIrrelevantNotifications()
+        // `didReceiveRemoteNotification` and`performFetchWithCompletionHandler` might be called if we're fetching instantly in foreground already
+        if appIsInForeground() {
+            logger.info("➡️ app already in foreground")
+            UserDefaults.pushToDebugArray("ABORT1")
+            completionHandler?(.newData)
+            return
+        }
 
-        // `didReceiveRemoteNotification` as well as `performFetchWithCompletionHandler` might be called if we're in foreground,
-        // in this case, there is no need to wait for things or do sth.
-        if appIsInForeground() || UserDefaults.nseFetching {
-            logger.info("➡️ app already in foreground or NSE running")
-            pushToDebugArray("OK1")
+        // abort if NSE is runnig; core cannot start I/O twice
+        if UserDefaults.nseFetching {
+            logger.info("➡️ NSE already running")
+            UserDefaults.pushToDebugArray("ABORT2")
```

Comment:
iirc, we tried that, and we were called less often as a result.

however, these tries were years ago, things may changed since then, but i needs research, and would not change it in this PR